test/Tester/ActivationServices/IStorageActivationService.cs

```diff
@@ -0,0 +1,78 @@
+﻿
+using Orleans.Runtime;
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Orleans;
+
+namespace Tester
+{
+    public interface IStorageActivationService<TState>
+    {
+        string Name { get; }
+
+        TState State { get; set; }
+
+        Task SaveAsync();
+
+        string GetExtendedInfo();
+    }
+
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public class StorageActivationServiceAttribute : Attribute
+    {
+        public string StateName { get; }
+        public string StorageProviderName { get; }
+
+        public StorageActivationServiceAttribute(string storageProviderName = null, string stateName = null)
+        {
+            this.StorageProviderName = storageProviderName;
+            this.StateName = stateName;
+        }
+    }
+
+    public class AttributedStorageActivationService<TState> : IStorageActivationService<TState>
+    {
+        private readonly IStorageActivationService<TState> storageActivationService;
+
+        public AttributedStorageActivationService(StorageActivationServiceFactory<TState> factory, IGrainActivationContext context)
+        {
+            var parameter = ActivationServiceUtilities.BindToConstructorParameter(context, typeof(IStorageActivationService<TState>));
```

Comment:
I agree more thought/work is needed here.  Will consider and attempt improvements in next pass.